daemon/cmd/hubble.go

```diff
@@ -162,12 +160,25 @@ func (d *Daemon) launchHubble() {
 			serveroption.WithHealthService(),
 			serveroption.WithObserverService(d.hubbleObserver),
 		}
-		tlsOption, err := buildHubbleTLSOption()
-		if err != nil {
-			logger.WithError(err).Error("Failed to initialize Hubble server")
-			return
+
+		// Hubble TLS/mTLS setup.
+		var tlsServerConfig *certloader.WatchedServerConfig
+		if option.Config.HubbleTLSDisabled {
+			options = append(options, serveroption.WithInsecure())
+		} else {
+			tlsServerConfigChan, err := certloader.EventualWatchedServerConfig(
+				logger.WithField("config", "tls-server"),
+				option.Config.HubbleTLSClientCAFiles,
+				option.Config.HubbleTLSCertFile,
+				option.Config.HubbleTLSKeyFile,
+			)
+			if err != nil {
+				logger.WithError(err).Error("Failed to initialize Hubble server TLS configuration")
+				return
+			}
+			tlsServerConfig = <-tlsServerConfigChan
```

Comment:
Before this patch `launchHubble()` would not block, and with this patch it will wait indefinitely for the TLS files to show up.

Question: Should we spin-up a goroutine to wait on the TLS files and then run the Hubble server so that `launchHubble()` won't block?

Reply:
Yes, either what you proposed or maybe just spawning `launchHubble()` itself in a go-routine.

Reply:
I've also tried to honor `d.ctx.Done()` in the select in case we're stuck on waiting for the files to show up. Let me know what you think!